### PR TITLE
chore(ci): automatically add new issues to the Design System project

### PR DESCRIPTION
**Pull request by tay1orjones**
Adds the [`actions/add-to-project`](https://github.com/marketplace/actions/add-to-github-projects-beta?version=v0.0.3) action that was [recently announced](https://github.blog/changelog/2022-04-07-the-new-github-issues-april-7th-update/) that will add new issues to the Design System project on issue creation.

#### Changelog

**New**

- added `actions/add-to-project`

#### Testing / Reviewing

- Once this is merged I'll create a new issue to test if it works

### CODE REVIEW

**Comment by tay1orjones** [comment]
FYI @sstrubberg @aagonzales @joshblack

File: .github/workflows/issue-triage.yml [hunk 1]
```diff
@@ -13,3 +13,11 @@ jobs:
...
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           enabled: true
+  add-to-project:
+  name: Add issue to Design System project
+  runs-on: ubuntu-latest
+  steps:
+    - uses: actions/add-to-project@v0.0.3
+      with:
+        project-url: https://github.com/orgs/carbon-design-system/projects/39
+        github-token: ${{ secrets.ADD_TO_PROJECT_PAT }}
```
Comment (joshblack) [inline .github/workflows/issue-triage.yml]: Would this section need to be indented for the options to be set on the `add-to-project` job?

Comment (tay1orjones) [reply .github/workflows/issue-triage.yml]: @joshblack I don't think so? [Docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-6) have it in line with `uses`, `with` doesn't have the dash though.

```yml
jobs:
  my_first_job:
    steps:
      - name: My first step
        uses: actions/hello_world@main
        with:
          first_name: Mona
          middle_name: The
          last_name: Octocat
```

Unless I'm misunderstanding what you're asking?

Comment (tay1orjones) [reply .github/workflows/issue-triage.yml]: Oh you mean up above, I see now in the full diff. Yeah I'll update 👍

Comment (tay1orjones) [reply .github/workflows/issue-triage.yml]: Fixed!

**Review by dakahn — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Pull request merged**